lib/rubocop/cop/lint/duplicate_require.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for duplicate `require`s and `require_relative`s.
+      #
+      # @example
+      #   # bad
+      #   require 'foo'
+      #   require 'bar'
+      #   require 'foo'
+      #
+      #   # good
+      #   require 'foo'
+      #   require 'bar'
+      #
+      class DuplicateRequire < Base
+        MSG = 'Duplicate `%<method>s` detected.'
+        REQUIRE_METHODS = %i[require require_relative].freeze
+
+        def_node_matcher :require_call?, <<~PATTERN
+          (send {nil? (const _ :Kernel)} {:#{REQUIRE_METHODS.join(' :')}} _)
+        PATTERN
+
+        def on_send(node)
+          return unless REQUIRE_METHODS.include?(node.method_name)
+
+          left_siblings_of(node).each do |sibling|
```

Comment:
This is `O(n^2)`, (`n` == number of requires), would it be hard to refactor as `O(n)` as it should be using a set?

Reply:
>  as it should be using a set?

Can you provide a hint on how can I rewrite to use this?

Reply:
Sure. I haven't tested, but I think that something like this should work:
```ruby
def on_new_investigation
  # Holds the known required files for a given parent node (used as key)
  @required = Hash.new{|h, k| h[k]=Set[] }.compare_by_identity
end

def on_send(node)
  return unless REQUIRE_METHODS.include?(node.method_name)

  next if @required[node.parent].add?(node.first_argument)
  add_offense(node, message: format(MSG, method: node.method_name))
end
```

Reply:
Updated with suggestion.